src/snovault/viewconfigs/searchview.py

```diff
@@ -0,0 +1,320 @@
+        self.used_filters = set_filters(self.request,
+                                        query,
+                                        self.result,
+                                        static_items)
+        # set facets
+        self.set_facets()
+
+        query['aggs'] = set_facets(self.facets,
+                                   self.used_filters,
+                                   self.principals,
+                                   self.doc_types)
+
+        # Search caching uses JSON as cache key, hence sorting the query
+        if sort:
+            query = sort_query(query)
+
+        return query
+
+    def set_facets(self):
+        if len(self.doc_types) == 1 and 'facets' in self.types[self.doc_types[0]].schema:
+            self.facets.extend(self.types[self.doc_types[0]].schema['facets'].items())
+        # Display all audits if logged in, or all but INTERNAL_ACTION if logged out
+        for audit_facet in audit_facets:
+            if self.search_audit and 'group.submitter' in self.principals or 'INTERNAL_ACTION' not in audit_facet[0]:
+                self.facets.append(audit_facet)
+
+    def construct_facets(self, es_results, total):
+        # Update facets
+        schemas = (self.types[item_type].schema for item_type in self.doc_types)
+        facets = format_facets(es_results,
+                               self.facets,
+                               self.used_filters,
+                               schemas,
+                               total,
+                               self.principals)
+        return facets
+
+    def query_elastic_search(self, query, es_index, from_=None, size=None):
+
+        # Execute the query
+        if size is None or size > 1000:
+            es_results = self.elastic_search.search(body=query,
+                                                    index=es_index,
+                                                    search_type='query_then_fetch')
+        else:
+            es_results = self.elastic_search.search(body=query,
+                                                    index=es_index,
+                                                    from_=from_,
+                                                    size=size,
+                                                    request_cache=True)
+        return es_results
+
+    def get_total_results(self, es_results):
+        total = es_results['hits']['total']
+        if not total:
+            # http://googlewebmastercentral.blogspot.com/2014/02/faceted-navigation-best-and-5-of-worst.html
+            self.request.response.status_code = 404
+            self.result['notification'] = 'No results found'
+            self.result['@graph'] = []
+        return total
+
+    def preprocess_view(self):
+        self.result['@id'] = '/search/' + self.search_base
+        self.result['@type'] = ['Search']
+        self.result['title'] = 'Search'
+        self.result['clear_filters'] = self.construct_clear_filters()
```

Comment:
construct_clear_filters is dependent on doc_types being build first.  Is looks like the current implementation builds the clear filters when self.doc_types = [].  

Moving self.construct_doc_types() above self.result['clear_filters'] = self.construct_clear_filters() is NOT the  solution.  The bottom code block in construct doc types needs to be below construct_clear_filters.  See that comment above.  The code needs to be untied;  We can't cleanly construct one before the other.

Reply:
doc_type is initialized in BaseView (base_view.py, line 24, code- "self.doc_types = self.request.params.getall('type')"). Where is self.doc_types set as an empty list?